doc/rtd/explanation/hardening.rst

```diff
@@ -0,0 +1,18 @@
+Security Hardening
+******************
+
+Cloud-init's use case is automating cloud instance initialization, with support
+across distributions and platforms. There is a myriad of ways to harden this
+space.
+
+Follow the security hardening guidelines provided by the OSes and cloud
+platforms that your cloud-init configuration is targeting.
+
+No plain text passwords
```

Comment:
Is it worth also calling out that it makes sense to ensure we have updated packages in your VM images upon launch to ensure latest fixes available are actually installed within a potentially stale cloud image? While Ubuntu cloud images are configured by default to enable unattended-upgrades (and resolve this issue overnight) it does leave a gap in current days launches that some software will remain out of date until packages are upgraded.

```
#cloud-config
package_upgrade: true 
```

We could link to Ubuntu's security guidelines https://documentation.ubuntu.com/server/explanation/security/security_suggestions/#keep-your-system-up-to-date

Reply:
> it makes sense to ensure we have updated packages in your VM images upon launch to ensure latest fixes available are actually installed within a potentially stale cloud image?

Recommending an update is fine, but I would suggest that we recommend using current images rather than assuming/accepting outdated ones. Launching an image with the latest packages is more secure than upgrading from an outdated image.

Reply:
I think reality is a bit in the middle, as (IIRC) there are clouds where the image publishing cadence is longer than months (like Oracle or IBM clouds).